dso/model.py

```diff
@@ -22,56 +24,94 @@ class ScanArtifact:
 
 class Datasource:
     WHITESOURCE = 'whitesource'
-    PROTECODE = 'protecode'
+    BDBA = 'bdba' # formerly protecode
     CHECKMARX = 'checkmarx'
     CLAMAV = 'clamav'
+    CC_UTILS = 'cc-utils'
 
 
-@dataclasses.dataclass
+@dataclasses.dataclass(frozen=True)
+class Artefact:
+    artefact_name: str
+    artefact_version: str
+    artefact_type: str
+    artefact_extra_id: dict
+
+
+@dataclasses.dataclass(frozen=True)
 class ArtefactReference:
-    componentName: str
-    componentVersion: str
-    artefact: typing.Union[
-        cm.Resource,
-        cm.ComponentSource,
-    ]
+    component_name: str
+    component_version: str
+    artefact: Artefact
 
 
-@dataclasses.dataclass
-class ComplianceMetadata:
+def artefact_ref_from_ocm(
+    component: cm.Component,
+    artefact: cm.Resource | cm.ComponentSource
+) -> ArtefactReference:
+    artefact = Artefact(
+        artefact_name=artefact.name,
+        artefact_version=artefact.version,
+        artefact_type=artefact.type,
+        artefact_extra_id=artefact.extraIdentity,
+    )
+    return ArtefactReference(
+        component_name=component.name,
+        component_version=component.version,
+        artefact=artefact,
+    )
+
+
+class Datatype:
+    VULNERABILITIES_AGGREGATED = 'vulnerabilities/aggregated'
+    VULNERABILITIES_RAW = 'vulnerabilities/raw'
+    MALWARE = 'malware'
+    LICENSES_AGGREGATED = 'licenses/aggregated'
+    COMPONENTS = 'components'
+    OS_IDS = 'os_ids'
+
+
+@dataclasses.dataclass(frozen=True)
+class Metadata:
     datasource: str
-    creationDate: typing.Union[datetime.datetime, str]
+    type: str
+    creation_date: datetime.datetime
 
 
-@dataclasses.dataclass
-class ComplianceData:
+@dataclasses.dataclass(frozen=True)
+class GreatestCVE:
+    greatestCvss3Score: typing.Optional[float]
+    reportUrl: str
+
+
+@dataclasses.dataclass(frozen=True)
+class OsID:
+    os_info: unixutil.model.OperatingSystemId
+
+
+@dataclasses.dataclass(frozen=True)
+class License:
```

Comment:
the name is rather misleading. why would a class named `License` (singular) have a field `licenses` (plural)? maybe rather call it `LicenseSummary`, or sth. the like? adding a short docstr could also help to make it easier to understand what this type represents